steering_controllers/steering_odometry/include/steering_odometry/steering_odometry.hpp

```diff
@@ -0,0 +1,197 @@
+
+  /**
+     * \brief Initialize the odometry
+     * \param time Current time
+     */
+  void init(const rclcpp::Time & time);
+
+  /**
+     * \brief Updates the odometry class with latest wheels position
+     * \param rear_wheel_pos  Rear wheel position [rad]
+     * \param front_steer_pos Front Steer position [rad]
+     * \param dt      time difference to last call
+     * \return true if the odometry is actually updated
+     */
+  bool update_from_position(
+    const double rear_wheel_pos, const double front_steer_pos, const double dt);
+
+  /**
+     * \brief Updates the odometry class with latest wheels position
+     * \param rear_wheel_vel  Rear wheel velocity [rad/s]
+     * \param front_steer_pos Front Steer position [rad]
+     * \param dt      time difference to last call
+     * \return true if the odometry is actually updated
+     */
+  bool update_from_velocity(
+    const double rear_wheel_vel, const double front_steer_pos, const double dt);
+
+  /**
+     * \brief Updates the odometry class with latest velocity command
+     * \param linear  Linear velocity [m/s]
+     * \param angular Angular velocity [rad/s]
+     * \param time    Current time
+     */
+  void update_open_loop(const double linear, const double angular, const double dt);
+
+  /**
+     * \brief heading getter
+     * \return heading [rad]
+     */
+  double get_heading() const { return heading_; }
+
+  /**
+     * \brief x position getter
+     * \return x position [m]
+     */
+  double get_x() const { return x_; }
+
+  /**
+     * \brief y position getter
+     * \return y position [m]
+     */
+  double get_y() const { return y_; }
+
+  /**
+     * \brief linear velocity getter
+     * \return linear velocity [m/s]
+     */
+  double get_linear() const { return linear_; }
+
+  /**
+     * \brief angular velocity getter
+     * \return angular velocity [rad/s]
+     */
+  double get_angular() const { return angular_; }
+
+  /**
+     * \brief Sets the wheel parameters: radius and separation
+     */
+  void set_wheel_params(double wheel_reparation_h, double wheel_radius, double wheelbase);
+
+  /**
+     * \brief Velocity rolling window size setter
+     * \param velocity_rolling_window_size Velocity rolling window size
+     */
+  void set_velocity_rolling_window_size(size_t velocity_rolling_window_size);
+
+  /**
+    * \brief TODO
+    * \param Vx  TODO
+    * \param theta_dot TODO
+    * \param wheelbase TODO
+    */
+  double convert_trans_rot_vel_to_steering_angle(double Vx, double theta_dot, double wheelbase);
+
+  /**
+     * \brief TODO
+     * \param Vx  TODO
+     * \param theta_dot TODO
+     */
+  std::tuple<double, double> twist_to_ackermann(double Vx, double theta_dot);
+
+  /**
+     *  \brief Reset poses, heading, and accumulators
+     */
+  void reset_odometry();
+
+private:
+  /// Rolling mean accumulator and window:
+  typedef rcpputils::RollingMeanAccumulator<double> RollingMeanAccumulator;
```

Comment:
We can probably live without this `typedef` and be explicit by the var definition.